### PR TITLE
ROX-19880: Secure notifiers on create and update

### PR DESCRIPTION
**Pull request by charmik-redhat**
## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
- [x] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: central/notifier/service/service.go [hunk 1]
```diff
@@ -29,11 +30,15 @@ type Service interface {
 func New(storage datastore.DataStore,
 	processor notifier.Processor,
 	policyCleaner policycleaner.PolicyCleaner,
-	reporter integrationhealth.Reporter) Service {
+	reporter integrationhealth.Reporter,
+	cryptoCodec cryptoutils.CryptoCodec,
```
Comment (clickboo) [inline central/notifier/service/service.go:34]: Let's make a singleton instantiated in central main?

File: central/notifier/service/service_impl.go [hunk 2]
```diff
@@ -186,7 +200,14 @@ func (s *serviceImpl) TestUpdatedNotifier(ctx context.Context, request *v1.Updat
 	if err := s.reconcileUpdateNotifierRequest(ctx, request); err != nil {
 		return nil, err
 	}
-	notifier, err := notifiers.CreateNotifier(request.GetNotifier())
+	if request.GetUpdatePassword() {
+		err := notifierUtils.SecureNotifier(request.GetNotifier(), s.cryptoKey)
+		if err != nil {
+			// Don't send out error from crypto lib
+			return nil, errors.New("Error securing notifier")
```
Comment (clickboo) [inline central/notifier/service/service_impl.go:207]: nit: error encrypting notifier credentials (can make this change with unit tests PR)

File: central/notifiers/utils/encryption.go [hunk 3]
```diff
@@ -18,3 +22,90 @@ func GetNotifierSecretEncryptionKey() (string, error) {
...
+	cryptoCodec := cryptocodec.Singleton()
+	var err error
+	switch notifier.GetType() {
+	case pkgNotifiers.JiraType:
+		jira := notifier.GetJira()
+		if jira == nil {
+			return nil
+		}
+		notifier.NotifierSecret, err = cryptoCodec.Encrypt(key, jira.GetPassword())
+		if err != nil {
```
Comment (clickboo) [inline central/notifiers/utils/encryption.go:43]: no need for if check, just return err should suffice

**Review by clickboo — APPROVED** [review]
lgtm, minor comments below

File: central/notifier/service/service_impl.go (hunk 2 above)
Comment (charmik-redhat) [reply central/notifier/service/service_impl.go:207]: I will make this change in a followup unit test PR.

**Pull request merged**